src/zscript/damnums/events.txt

```diff
@@ -6,43 +6,125 @@
 
 class DamNumEventHandler : StaticEventHandler
 {
-	Array<DamNumAccumulator> accumulators;
-
 	/*
 	 * Universal DamNum spawner. If it's a monster, throw some nums!
 	 */
+	
 	override void WorldThingDamaged(WorldEvent e)
 	{
-		if(e.thing && e.Thing.bISMONSTER && e.DamageSource)
+		if (dam_enabled && e.thing && e.Thing.bISMONSTER && e.DamageSource)
 		{
 			// Just spawn the numbers directly in "shotgun spray" mode;
 			// otherwise, fire up the accumulator.
-			if(dam_spray) {
+			let dni = DamNumInv(e.Thing.FindInventory("DamNumInv"));
+			if (dni)
+			{
+				let tdk = dni.dk;
+				if (tdk && tdk.thing)
+					tdk.position = tdk.thing.pos + (0,0,tdk.thing.height);
```

Comment:
Quick request #1: Would you mind at all adding the line `tdk.damagetype = e.DamageType` here? Just so I don't forget later when I get around to implementing it.